src/features/roles-and-permissions/modals/CreateNewRoleModal.tsx

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useState } from "react";
+import { Flex, TextField } from "@radix-ui/themes";
+
+import { Modal } from "@shared/components/modal";
+import Form from "@shared/components/form";
+import ItemLabel from "@shared/components/itemLabel";
+import Button from "@shared/components/button";
+import { AssignPermission } from "@shared/components/assignPermission";
+
+import { useRolesList } from "../hooks";
+
+interface CreateNewRoleModalProps {
```

Comment:
I'd prefer if we didn't use interfaces at all
```suggestion
type CreateNewRoleModalProps = {
```

Reply:
Updated